openml/datasets/dataset.py

```diff
@@ -303,28 +347,100 @@ def decode_arff(fh):
             with io.open(filename, encoding='utf8') as fh:
                 return decode_arff(fh)
 
+    @staticmethod
+    def _convert_array_format(data, array_format, attribute_names):
```

Comment:
Maybe document supported conversions and how they ought to be specified. From reading code, it looks like there are only two specific conversions allowed:
 - non-sparse numeric data to numpy-array (`array_format="array"`)
 - sparse data to sparse dataframe (`array_format="dataframe"`)

no cross combinations are possible (sparse to non-sparse and vice versa).
Additionally it seems the method converts categorical columns by using their numeric representation? It might be good to document that as well.